### ISSUE TITLE
Unused steam hot water is not dumped in some scenarios

### ISSUE THREAD

**Issue by michieldenhaan** [quintel/etengine#1173]
Scenario: 1435023 (BETA)

Demand of `energy_heat_unused_steam_hot_water` is positive but demand of `energy_heat_dumped_steam_hot_water` is zero, even though it's the only node connecting into `energy_heat_unused_steam_hot_water`.

<img width="1355" alt="image" src="https://user-images.githubusercontent.com/32056448/118850756-62313600-b8d1-11eb-8bc5-14dfc8f89c71.png">

@antw Do you have time to look into this?

**Comment by michieldenhaan** [comment]
Right after hitting 'Create issue' I noticed that the loss of this node is 100% 🤔 This seems to be the case for all derived datasets I've checked but not for parent datasets..

**Comment by antw** [comment]
I think this is because of the inversed flexible edge to the "dump" node. In Refinery, the share of carriers (loss, steam hot water) comes either from shares provided by the source data (in a node file), or is calculated dynamically based on the demand of the edges.

For example:

```
 [ Loss Node ]  <- 25MJ --
                            [ Unused heat: 100MJ ]
[ Steam Node ]  <- 75MJ --
```

If there is 100MJ of unused heat flowing into 25MJ of loss and 75MJ of steam, it's easy to determine that the shares are 0.25 and 0.75 respectively. However, when there's no demand Refinery would not be able to calculate the share of each carrier.

This would be a problem – particularly with import / export, or where there are loops in the graph – where a dataset has no demand. I think [hydrogen is a good example](https://engine.energytransitionmodel.com/inspect/835777/graphs/energy/nodes/energy_distribution_hydrogen_before_transport). We have import coming from *reversed* flexible edges, and export leaving via an *inversed* flexible edge.

Inversed flexible edges therefore have special behaviour in Refinery: when the node demand is zero and the edge has a sibling, the inversed flexible share is set to zero. This allows Refinery to calculate the graph, while also allowing ETEngine to do the right thing and calculate the inversed flexible itself based on the energy flows.

To opt out of the special behaviour, the inversed flexible needs to be changed to a reversed flexible:

```diff
- - type = inversed_flexible
+ - type = flexible
+ - reversed = true
```

... and the unused_steam_hot_water node needs to be provided with a default conversion for steam_hot_water:

```diff
- - output.loss = elastic
+ - output.loss = 0.0  # This can stay "elastic", but I think 0.0 is clearer.
+ - output.steam_hot_water = 1.0
```

This should fix the problem for datasets where there's no steam hot water while keeping the same results for full-size datasets.

**Comment by michieldenhaan** [comment]
Thanks! I've sent you a PR.
